spec/system/user_show_spec.rb

```diff
@@ -0,0 +1,65 @@
+RSpec.describe 'User profile page', type: :feature do
+  before(:each) do
+    @user = User.create(
+      name: 'John Doe',
+      bio: 'Web Developer',
+      photo: 'https://via.placeholder.com/150',
+      posts_counter: 0
+    )
+
+    # Create first 3 posts for this user
+    3.times do |n|
+      @post = @user.posts.create(title: "Title #{n + 1}", text: 'Lorem ipsum dolor sit amet.', comments_counter: 0, author_id: 1, likes_counter: 0)
+    end
+  end
+
+  it 'shows user profile picture' do
+    visit user_path(@user)
+    expect(page).to have_css("img[src*='https://via.placeholder.com/150']")
+  end
+
+  it "shows user's username" do
+    visit user_path(@user)
+    expect(page).to have_content('John Doe')
+  end
+
+  it 'shows the number of posts the user has written' do
+    visit user_path(@user)
+    expect(page).to have_content('Number of posts: 3')
+  end
+
+  it "shows the user's bio" do
+    visit user_path(@user)
+    expect(page).to have_content('Web Developer')
+  end
+
+  it "shows the user's first 3 posts" do
+    visit user_path(@user)
+
+    # Check that the titles of the first 3 posts are present on the page
+    expect(page).to have_content('Title 1')
+    expect(page).to have_content('Title 2')
+    expect(page).to have_content('Title 3')
+  end
+
+  it 'displays a button for all user posts' do
+    visit user_path(@user)
+    expect(page).to have_content('See all posts')
+  end
+
+  it 'click a user post, it redirects to that specific post show page' do
+    # navigate to the user profile page
+    visit user_path(@user)
+    # click the "See all posts" button
+    click_link 'See all posts'
+    # assert that the current URL is the expected one
+    expect(page).to have_current_path(user_posts_path(@user))
+    expect(page).to have_content('Lorem ipsum dolor sit amet.')
+  end
+
+  it 'click to see all posts, it redirects to the user post index page' do
+    visit user_path(@user)
+    click_link 'See all posts'
+    expect(page).to have_current_path(user_posts_path(@user))
+  end
+end
```

Comment:
- [ ] Two of the failures come from the `user_show_spec.rb` 

```bash
18) User profile page click a user post, it redirects to that specific post show page
      Failure/Error: <% @posts.each do |post| %>
      
      ActionView::Template::Error:
        PG::UndefinedColumn: ERROR:  column likes.post_id does not exist
        LINE 1: SELECT "likes".* FROM "likes" WHERE "likes"."post_id" IN ($1...
                                                    ^
      # ./app/views/posts/index.html.erb:9:in `_app_views_posts_index_html_erb__2018209907907721745_28180'
      # ./spec/system/user_show_spec.rb:56:in `block (2 levels) in <main>'
      # ------------------
      # --- Caused by: ---
      # PG::UndefinedColumn:
      #   ERROR:  column likes.post_id does not exist
      #   LINE 1: SELECT "likes".* FROM "likes" WHERE "likes"."post_id" IN ($1...
      #                                               ^
      #   ./app/views/posts/index.html.erb:9:in `_app_views_posts_index_html_erb__2018209907907721745_28180'

  19) User profile page click to see all posts, it redirects to the user post index page
      Failure/Error: <% @posts.each do |post| %>
      
      ActionView::Template::Error:
        PG::UndefinedColumn: ERROR:  column likes.post_id does not exist
        LINE 1: SELECT "likes".* FROM "likes" WHERE "likes"."post_id" IN ($1...
                                                    ^
      # ./app/views/posts/index.html.erb:9:in `_app_views_posts_index_html_erb__2018209907907721745_28180'
      # ./spec/system/user_show_spec.rb:64:in `block (2 levels) in <main>'
      # ------------------
      # --- Caused by: ---
      # PG::UndefinedColumn:
      #   ERROR:  column likes.post_id does not exist
      #   LINE 1: SELECT "likes".* FROM "likes" WHERE "likes"."post_id" IN ($1...
      #                                               ^
      #   ./app/views/posts/index.html.erb:9:in `_app_views_posts_index_html_erb__2018209907907721745_28180'
```

Here's what you can do:

Check the database schema: Verify that the likes table has a post_id column. If it doesn't exist, you need to create a migration to add it. You can use the following command in your terminal to generate the migration file:

`rails generate migration AddPostIdToLikes post:references`

This command will generate a migration file with the necessary changes to add the post_id column to the likes table.

Run the migration: Apply the migration to your database by running the following command in your terminal:

`rails db:migrate`

This will execute the migration and add the post_id column to the likes table.

Update the Like model: In the Like model, make sure you have the proper association with the Post model. It should include a belongs_to :post declaration.


```ruby
class Like < ApplicationRecord
  belongs_to :post
end
```
This association ensures that a Like record is associated with a Post record through the post_id foreign key.

After making these changes, run your tests again to verify if the errors related to the missing post_id column in the likes table are resolved.

- **[OPTIONAL]** The rest of the tests are related to models and controller folders. Please 🙏🏽 fix them to pass all the tests 👇🏽 

```bash
 20) Comment callbacks increments the post's comment counter after a new comment is created
      Failure/Error: Comment.create(author_id: user.id, post_id: post.id)
      
      ActiveModel::UnknownAttributeError:
        unknown attribute 'author_id' for Comment.
      # ./spec/models/comment_spec.rb:10:in `block (4 levels) in <main>'
      # ./spec/models/comment_spec.rb:9:in `block (3 levels) in <main>'

  21) Like callbacks increments the post's likes counter after a new like is created
      Failure/Error: Like.create(author_id: user.id, post_id: post.id)
      
      ActiveModel::UnknownAttributeError:
        unknown attribute 'post_id' for Like.
      # ./spec/models/like_spec.rb:10:in `block (4 levels) in <main>'
      # ./spec/models/like_spec.rb:9:in `block (3 levels) in <main>'

  22) Post recent_comments method returns up to 5 recent comments for the post
      Failure/Error: post.comments.create(author_id: user.id, text: 'Comment 1')
      
      ActiveModel::UnknownAttributeError:
        unknown attribute 'author_id' for Comment.
      # ./spec/models/post_spec.rb:58:in `block (3 levels) in <main>'
      # ./spec/models/post_spec.rb:65:in `block (3 levels) in <main>'

  23) Post recent_comments method returns less than or equal to 5 comments if there are fewer than 5 comments for the post
      Failure/Error: post.comments.create(author_id: user.id, text: 'Comment 1')
      
      ActiveModel::UnknownAttributeError:
        unknown attribute 'author_id' for Comment.
      # ./spec/models/post_spec.rb:58:in `block (3 levels) in <main>'
      # ./spec/models/post_spec.rb:69:in `block (3 levels) in <main>'

  24) Post recent_comments method returns an empty array if there are no comments for the post
      Failure/Error: post.comments.create(author_id: user.id, text: 'Comment 1')
      
      ActiveModel::UnknownAttributeError:
        unknown attribute 'author_id' for Comment.
      # ./spec/models/post_spec.rb:58:in `block (3 levels) in <main>'
      # ./spec/models/post_spec.rb:76:in `block (3 levels) in <main>'

  25) Posts GET #show returns a successful response
      Failure/Error: @post = Post.includes(:comments, :likes, :author).find(params[:id])
      
      ActiveRecord::StatementInvalid:
        PG::UndefinedColumn: ERROR:  column likes.post_id does not exist
        LINE 1: SELECT "likes".* FROM "likes" WHERE "likes"."post_id" = $1
                                                    ^
      # ./app/controllers/posts_controller.rb:14:in `show'
      # ./spec/requests/posts_spec.rb:34:in `block (3 levels) in <main>'
      # ------------------
      # --- Caused by: ---
      # PG::UndefinedColumn:
      #   ERROR:  column likes.post_id does not exist
      #   LINE 1: SELECT "likes".* FROM "likes" WHERE "likes"."post_id" = $1
      #                                               ^
      #   ./app/controllers/posts_controller.rb:14:in `show'

  26) Posts GET #show assigns the correct user and post
      Failure/Error: @post = Post.includes(:comments, :likes, :author).find(params[:id])
      
      ActiveRecord::StatementInvalid:
        PG::UndefinedColumn: ERROR:  column likes.post_id does not exist
        LINE 1: SELECT "likes".* FROM "likes" WHERE "likes"."post_id" = $1
                                                    ^
      # ./app/controllers/posts_controller.rb:14:in `show'
      # ./spec/requests/posts_spec.rb:39:in `block (3 levels) in <main>'
      # ------------------
      # --- Caused by: ---
      # PG::UndefinedColumn:
      #   ERROR:  column likes.post_id does not exist
      #   LINE 1: SELECT "likes".* FROM "likes" WHERE "likes"."post_id" = $1
      #                                               ^
      #   ./app/controllers/posts_controller.rb:14:in `show'

  27) Posts GET #show has a correct status
      Failure/Error: @post = Post.includes(:comments, :likes, :author).find(params[:id])
      
      ActiveRecord::StatementInvalid:
        PG::UndefinedColumn: ERROR:  column likes.post_id does not exist
        LINE 1: SELECT "likes".* FROM "likes" WHERE "likes"."post_id" = $1
                                                    ^
      # ./app/controllers/posts_controller.rb:14:in `show'
      # ./spec/requests/posts_spec.rb:45:in `block (3 levels) in <main>'
      # ------------------
      # --- Caused by: ---
      # PG::UndefinedColumn:
      #   ERROR:  column likes.post_id does not exist
      #   LINE 1: SELECT "likes".* FROM "likes" WHERE "likes"."post_id" = $1
      #                                               ^
      #   ./app/controllers/posts_controller.rb:14:in `show'

Finished in 1.75 seconds (files took 2.46 seconds to load)
61 examples, 27 failures
```

Reply:
Dear @franclobo , really thanks for your comprehensive reviews 🙏